src/aks-preview/azext_aks_preview/agentpool_decorator.py

```diff
@@ -287,7 +287,8 @@ def _get_disable_windows_outbound_nat(self) -> bool:
         if self.decorator_mode == DecoratorMode.CREATE:
             if (
                 self.agentpool and
-                self.agentpool.windows_profile and
+                hasattr(self.agentpool, "windows_profile") and self.agentpool.windows_profile and
+                hasattr(self.agentpool.windows_profile, "disable_windows_outbound_nat") and
                 self.agentpool.windows_profile.disable_windows_outbound_nat is not None
```

Comment:
```suggestion
                self.agentpool.windows_profile.disable_outbound_nat is not None
```
should correct the property name

Reply:
Updated. Thank you.